wallet-core/tests/notes.rs

```diff
@@ -48,6 +48,19 @@ pub fn gen_note<T: RngCore + CryptoRng>(
     }
 }
 
+/// Generate a note leaf, useful for testing purposes
```

Comment:
Moana mentioned she is against exposing any function that is just used for tests, i moved this initially into the tests folder

Reply:
And she is correct, but this function is part of the tests, is never exposed to anything, it's not on the library. To be fair, it has the `pub` only because I copied the existing one, `gen_note`, that has the same visibility identifier.

TBH we could remove both of them, but there is also no harm to keep them: as said, those functions are just defined in tests, not in the library.